File: backend/app/schemas/camera_coverage.py
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from app.core.camera_coverage import MAX_CAMERA_COVERAGE_RANGE_METRES


class CameraCoverageInput(BaseModel):
    origin_latitude: float = Field(..., ge=-90, le=90)
    origin_longitude: float = Field(..., ge=-180, le=180)
    coverage_bearing_degrees: float = Field(..., ge=0, lt=360)
    coverage_angle_degrees: float = Field(..., ge=1, le=180)
    coverage_range_metres: float = Field(
        ...,
        ge=1,
        le=MAX_CAMERA_COVERAGE_RANGE_METRES,
    )


class CameraCoverageResponse(CameraCoverageInput):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    camera_id: UUID
    polygon: list[list[float]]

File: frontend/src/lib/validators/camera-coverage.ts
import { z } from "zod";

export const MAX_CAMERA_ORIGIN_DISTANCE_METRES = 100;
export const MAX_CAMERA_COVERAGE_RANGE_METRES = 200;

export const cameraCoverageSchema = z.object({
  origin_latitude: z.number().finite().min(-90).max(90),
  origin_longitude: z.number().finite().min(-180).max(180),
  coverage_bearing_degrees: z.number().finite().min(0).lt(360),
  coverage_angle_degrees: z.number().finite().min(1).max(180),
  coverage_range_metres: z
    .number()
    .finite()
    .min(1)
    .max(MAX_CAMERA_COVERAGE_RANGE_METRES),
});

export type CameraCoverageInput = z.infer<typeof cameraCoverageSchema>;